internal/datatransfer/monitor.go

```diff
@@ -76,7 +81,13 @@ func (m *Monitor) syncPathsWorkload(workloadName string) {
 		return
 	}
 
-	s3sync, err := s3.NewSync(*storage.S3)
+	syncWrapper, err := m.getFsSync()
+	if err != nil {
+		log.Errorf("Error while getting s3 synchronizer: %v", err)
+		return
+	}
+
+	err = syncWrapper.Connect()
```

Comment:
What will happen if `Connect` is called here while `SyncPath` is being executed in `syncPaths`? Will there be any problem?

Reply:
No, I do not think so, the pointer is copy, so shouldn't be a problem at all. Or I cannot see it.

Reply:
The pointer is a copy of pointer stored in the struct, but it points at the same value so effectively `Connect` is called against the same object, which changes internal state of `syncWrapper`. But is that dangerous in any way?

Reply:
Added a if on the s3/sync.go

https://github.com/jakub-dzon/k4e-device-worker/pull/38/files#diff-5a3f35566e47eb15cffc1c02a0b18f6bbbf48883663884e699ed09ca12942ca7R60